src/stats/actions.ts

```diff
@@ -152,60 +142,92 @@ function handleActionCompute(state: PlayerActionState, indices: PlayerIndexedTyp
   // Manage animation state
   const currentAnimation = playerFrame.actionStateId!;
   state.animations.push(currentAnimation);
+  const currentFrameCounter = playerFrame.actionStateCounter!;
+  state.actionFrameCounters.push(currentFrameCounter);
 
   // Grab last 3 frames
   const last3Frames = state.animations.slice(-3);
   const prevAnimation = last3Frames[last3Frames.length - 2] as number;
-  const newAnimation = currentAnimation !== prevAnimation;
+  const prevFrameCounter = state.actionFrameCounters[state.actionFrameCounters.length - 2] as number;
+
+  // New action if new animation or frame counter goes back down (repeated action)
+  const isNewAction =
+    currentAnimation !== prevAnimation ||
+    (currentAnimation === prevAnimation && prevFrameCounter > currentFrameCounter);
```

Comment:
```suggestion
  const isNewAction = currentAnimation !== prevAnimation || prevFrameCounter > currentFrameCounter;
```
Pretty sure if `currentAnimation !== prevAnimation` is false, then the next clause guarantees it to be true, thus you don't have to compare it twice.